docs/theme-context.rst

```diff
@@ -0,0 +1,123 @@
+Read the Docs data passed to Sphinx build context
+=================================================
+
+Before calling `sphinx-build` to render your docs, Read the Docs injects some
+extra context in the templates by using the `html_context Sphinx setting`_ in
+the ``conf.py`` file. This extra context allows Read the Docs to add some
+features like "Edit on GitHub" and use a user custom Analytics code, among others.
+
+.. _html_context Sphinx setting: http://www.sphinx-doc.org/en/stable/config.html#confval-html_context
+
+Context injected by default
+---------------------------
+
+Here is the full list of values injected by Read the Docs as a Python
+dictionary. Note that this dictionary is injected under the main key
+`readthedocs`:
+
+
+.. code:: python
+
+    'readthedocs': {
+        'version': {
+            'pk': int,
+            'name': str,
+            'slug': str,
+            'downloads': {
+                'pdf: str,
+                'htmlzip': str,
+                'epub': str
+            },
+            'resource_uri': '/api/v2/version/{pk}/'
+        },
+        'project': {
+            'pk': int
+            'name': str,
+            'slug': str,
+            'canonical_url': str,
+            'resource_uri': '/api/v2/project/{pk}/'
+        },
+        'sphinx': {
+            'html_theme': str,
+            'source_suffix': str,
+        },
+        'analytics': {
+            'user_analytics_code': str,
+            'global_analytics_code': str
+        },
+        'vcs': {
+            'type': str,  # 'bitbucket', 'github' or 'gitlab'
+            'user': str,
+            'repo': str,
+            'commit': str,
+            'version': str,
+            'display': bool,
+            'conf_py_path': str
+        },
+        'meta': {
+            'API_HOST': str,
+            'MEDIA_URL': str, 
+            'PRODUCTION_DOMAIN': str,
+            'READTHEDOCS': True
+        }
```

Comment:
This is missing `subprojects`

Reply:
Mjm, I finally removed the `subprojects` because I supposed this was dynamic data that should be retrieved by JS. I wasn't really sure what to do here.

Isn't _"dangerous"_ to make it static under this context? Won't this give us some problems in the future?

Reply:
I'm still not convinced on this argument about injecting dynamic data. It still seems like something that users could want for various reasons in their doc context, without having to render it via JS/API. "It might go out of date when a new version is built" doesn't seem like a compelling enough reason to not include the live data when the build happens. What if they just want to include a subset of the data that they know will always be valid, for example? Just because _our_ user case requires fresh data, doesn't meal all use cases do. 

We should _warn_ users of the possible downsides of using this data, but not passing it in feels wrong to me.

Reply:
Good!

I re-structure the notes to be warnings and communicate the user some of the problems of using this static data and what would be the solution (using the API via JS). I wrote the idea, probably the English could be improved.

Reply:
I also added the `subprojects` to the `project` object; I'm not sure if I have to add the `versions` also under this object or not.